### PR TITLE
Make sure num_segments is at least 1

### LINKED ISSUE #1272: Crash when rendering MultiMeterStatusBar when values are zero
### Environment

* **FreeOrion Version**: 0.4.6
* **Operating System**: Linux (Fedora 25)
* **Graphic card used**: Intel HD 530
* **Fetched as**: Compiled from source

### Description
I'm the package maintainer for freeorion in Fedora. I recently received a [crash report](https://bugzilla.redhat.com/show_bug.cgi?id=1416197) that looks like it's caused by an implicit cast from a signed int to an unsigned int (size_t). `length_error` is thrown by std::vector when `reserve` is called and the passed in `size_t` is larger than the size of the vector. That got me to inspect the values as computed inside `MultiMeterStatusBar.cpp` and noticed that when `num_segments` works out to be `0` (as is the case when all the resources of an Outpost on a planet are zero), when `bar_verts.reserve(num_segments - 1)` is called, the expression `num_segments - 1` evaluates to `-1`, but `reserve` implicitly casts this to an unsigned `size_t`. In the debugger, I was able to see these values are huge.

```
(lldb) p bar_verts.b_data.max_size()
(size_type) $5 = 4611686018427387903
(lldb) p (size_type) num_segments - 1
(unsigned long) $6 = 18446744073709551615
```

### Expected Result
The resource status bar to pull down.

### Actual Result
Crash, throwing `length_error`.

### Steps to reproduce
1. Build Outpost ship
2. Deploy the outpost on some planet
3. Click on pull-down icon in outpost detail window ( prod/research/trade/supply where all 4 values are 0 )

### PR DESCRIPTION
**Pull request by subpop**
After computing the possible range for the meter, make sure num_segments is 1 before passing it into the reserve method.

This patch fixes this particular crash, but I'm not sure if it's the best place to fix it. I didn't fix it inside `GG/GLClientAndServerBuffer.cpp` since it felt like as far as `GLClientAndServerBufferBase` is concerned, it *should* assume the parameter passed to `reserve` is unsigned.

fixes #1272

### CODE REVIEW

File: UI/MultiMeterStatusBar.cpp [hunk 1]
```diff
@@ -128,6 +128,8 @@ void MultiMeterStatusBar::Render() {
     // lines for 20, 40, 60, 80 etc.
     int num_segments = num_full_increments * 5;
     GG::GL2DVertexBuffer bar_verts;
+    if (num_segments == 0)
```
Comment (geoffthemedio) [inline UI/MultiMeterStatusBar.cpp:131]: I'd rephrase this as something like `num_segments = std::max(1, num_segments);`

**Comment by Vezzra** [comment]
Another PR providing a fix (PR #1274) has been merged, so this one is obsolete. Tagging as "duplicate".

**Comment by LGM-Doyle** [comment]
@subpop sorry about adding a PR after you had already contributed a fix.  I checked that no one assigned themselves to the issue, but not if someone had posted a PR that linked back to the issue.

**Comment by subpop** [comment]
No problem. I'm fine with anyone patching it who's more familiar with the code base. I'm going to actually update my patch in my package to be based off #1274. 👍

**Comment by LGM-Doyle** [comment]
@subpop, I hope that you mean to use the single commit from PR #1274, cherry picked onto the 0.4.6 branch that you have already packaged.  

If you use #1274 directly, it is based on current master and includes the incomplete fighters work and new known unfixed bugs.

**Comment by subpop** [comment]
Oh yea. I pulled just this commit into a patch (git format-patch) and applied it to the 0.4.6 tarball.

http://pkgs.fedoraproject.org/cgit/rpms/freeorion.git/commit/?id=ee1a614f70d59d1b26fa54d02f874fd5b3057674
